tests/test_main.py

```
@@ -81,6 +82,99 @@ def _mock_run_with_keyboard_interrupt(coro: Any) -> None:
+        super().__init__()
+        self.cancel_called = False
+
+    def cancel(self) -> bool:
+        self.cancel_called = True
+        return super().cancel()
+
+
+class _TimeoutCloseFuture(_CloseFutureBase):
+    """Future that raises TimeoutError immediately on result()."""
+
+    def result(self, timeout: float | None = None) -> None:
```

Ruff (0.14.13)

Unused method argument: `timeout`

(ARG002)